### PR TITLE
Enable file module to disable diff_mode

### LINKED ISSUE #80817: ansible.builtin.file diff could not be disabled
### Summary

I'm trying to desactivate diff in ansible.builtin.file task like below while using the diff_mode (ansible-playbook playbook.yml --diff)  : 

```
- name: "Create Folder ../artifacts/{{ inventory_hostname }}/ if not exist"
  ansible.builtin.file:
    path: "../artifacts/{{ inventory_hostname }}"
    mode: 0755
    state: directory
  delegate_to: localhost
  diff: false
```
The option "diff: false" seams not beeing implemented in file module.

**The output is :**
```
TASK [orange.vpn : Create Folder ../artifacts/NEXUS-93180-1/ if not exist] **************************************************************************************************************************************************************************************************************************************************************
--- before
+++ after
@@ -1,4 +1,4 @@
 {
-    "mode": "0777",
+    "mode": "0755",
     "path": "../artifacts/NEXUS-93180-1"
 }
```

**It should be :** 
```
TASK [orange.vpn : Create Folder ../artifacts/NEXUS-93180-1/ if not exist] **************************************************************************************************************************************************************************************************************************************************************
changed: [NEXUS-93180-1 -> localhost]
```

### Issue Type

Feature Idea

### Component Name

ansible.builtin.file

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by cavcrosby**
##### SUMMARY

Resolves #80817.

##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

File: lib/ansible/modules/file.py [hunk 1]
```diff
@@ -983,6 +983,9 @@ def main():
     elif state == 'absent':
         result = ensure_absent(path)
 
+    if not module._diff:
```
Comment (Akasurde) [inline lib/ansible/modules/file.py:986]: Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Comment by cavcrosby** [comment]
@Akasurde any other thoughts or critiques on my PR?

**Review by Akasurde — APPROVED** [review]

**Pull request merged**